### ISSUE TITLE
Ubuntu 20.04 certbot error

### ISSUE THREAD

**Issue by DatGuyMcCaully** [LemmyNet/lemmy-ansible#89]
Looks like I'm getting a python error from certbot, running Ubuntu Linux (ubuntu 20.04), fresh install of Linux, have only ran the stuff from this repository. 

Removed domain and email for privacy

TASK [request initial letsencrypt certificate] ***************************************************************************************************
fatal: [lemmy.ristl.org]: FAILED! => {"changed": true, "cmd": ["certbot", "certonly", "--nginx", "--agree-tos", "--cert-name", "<domain>", "-d", ""<domain>", "-m", "<email>"], "delta": "0:00:00.353763", "end": "2023-06-21 14:18:18.269786", "msg": "non-zero return code", "rc": 1, "start": "2023-06-21 14:18:17.916023", "stderr": "Traceback (most recent call last):\n  File \"/usr/lib/python3/dist-packages/requests_toolbelt/_compat.py\", line 49, in <module>\n    from requests.packages.urllib3.contrib import appengine as gaecontrib\nImportError: cannot import name 'appengine' from 'requests.packages.urllib3.contrib' (/usr/local/lib/python3.8/dist-packages/urllib3/contrib/__init__.py)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/bin/certbot\", line 11, in <module>\n    load_entry_point('certbot==0.40.0', 'console_scripts', 'certbot')()\n  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 490, in load_entry_point\n    return get_distribution(dist).load_entry_point(group, name)\n  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2854, in load_entry_point\n    return ep.load()\n  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2445, in load\n    return self.resolve()\n  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2451, in resolve\n    module = __import__(self.module_name, fromlist=['__name__'], level=0)\n  File \"/usr/lib/python3/dist-packages/certbot/main.py\", line 20, in <module>\n    from certbot import client\n  File \"/usr/lib/python3/dist-packages/certbot/client.py\", line 14, in <module>\n    from acme import client as acme_client\n  File \"/usr/lib/python3/dist-packages/acme/client.py\", line 16, in <module>\n    from requests_toolbelt.adapters.source import SourceAddressAdapter\n  File \"/usr/lib/python3/dist-packages/requests_toolbelt/__init__.py\", line 12, in <module>\n    from .adapters import SSLAdapter, SourceAddressAdapter\n  File \"/usr/lib/python3/dist-packages/requests_toolbelt/adapters/__init__.py\", line 12, in <module>\n    from .ssl import SSLAdapter\n  File \"/usr/lib/python3/dist-packages/requests_toolbelt/adapters/ssl.py\", line 16, in <module>\n    from .._compat import poolmanager\n  File \"/usr/lib/python3/dist-packages/requests_toolbelt/_compat.py\", line 51, in <module>\n    from urllib3.contrib import appengine as gaecontrib\nImportError: cannot import name 'appengine' from 'urllib3.contrib' (/usr/local/lib/python3.8/dist-packages/urllib3/contrib/__init__.py)", "stderr_lines": ["Traceback (most recent call last):", "  File \"/usr/lib/python3/dist-packages/requests_toolbelt/_compat.py\", line 49, in <module>", "    from requests.packages.urllib3.contrib import appengine as gaecontrib", "ImportError: cannot import name 'appengine' from 'requests.packages.urllib3.contrib' (/usr/local/lib/python3.8/dist-packages/urllib3/contrib/__init__.py)", "", "During handling of the above exception, another exception occurred:", "", "Traceback (most recent call last):", "  File \"/usr/bin/certbot\", line 11, in <module>", "    load_entry_point('certbot==0.40.0', 'console_scripts', 'certbot')()", "  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 490, in load_entry_point", "    return get_distribution(dist).load_entry_point(group, name)", "  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2854, in load_entry_point", "    return ep.load()", "  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2445, in load", "    return self.resolve()", "  File \"/usr/lib/python3/dist-packages/pkg_resources/__init__.py\", line 2451, in resolve", "    module = __import__(self.module_name, fromlist=['__name__'], level=0)", "  File \"/usr/lib/python3/dist-packages/certbot/main.py\", line 20, in <module>", "    from certbot import client", "  File \"/usr/lib/python3/dist-packages/certbot/client.py\", line 14, in <module>", "    from acme import client as acme_client", "  File \"/usr/lib/python3/dist-packages/acme/client.py\", line 16, in <module>", "    from requests_toolbelt.adapters.source import SourceAddressAdapter", "  File \"/usr/lib/python3/dist-packages/requests_toolbelt/__init__.py\", line 12, in <module>", "    from .adapters import SSLAdapter, SourceAddressAdapter", "  File \"/usr/lib/python3/dist-packages/requests_toolbelt/adapters/__init__.py\", line 12, in <module>", "    from .ssl import SSLAdapter", "  File \"/usr/lib/python3/dist-packages/requests_toolbelt/adapters/ssl.py\", line 16, in <module>", "    from .._compat import poolmanager", "  File \"/usr/lib/python3/dist-packages/requests_toolbelt/_compat.py\", line 51, in <module>", "    from urllib3.contrib import appengine as gaecontrib", "ImportError: cannot import name 'appengine' from 'urllib3.contrib' (/usr/local/lib/python3.8/dist-packages/urllib3/contrib/__init__.py)"], "stdout": "", "stdout_lines": []}

**Comment by gribodyr** [comment]
Same for me. Running Ubuntu Linux (ubuntu 20.04 with all package upgrades), fresh install of Linux,.

**Comment by DatGuyMcCaully** [comment]
installed the following version of python on 20.04: python3, python3.8, python3.11 and uncommited the following in the ansible.cfg:
`#interpreter_python= /usr/bin/python3`
and changed it to 
`interpreter_python= /usr/bin/python3.*`
the * being the python version.
none worked.

Also, spun up a new azure server with Ubuntu 22.04 and still running into the same issues.

**Comment by christopher-conley** [comment]
@DatGuyMcCaully

It's due to the version of urllib3 distributed with the "python3-pip" in Ubuntu 20.04 combined with the version that the docker-compose pip module wants. The version of urllib3 that gets installed alongside "python3-pip" is 1.25.8, whereas the docker-compose pip module wants something that is >=1.26.0. When the docker-compose pip module gets installed, to meet that requirement, pip installs a newer version of urllib3 which seems to be bugged ([see here](https://github.com/apache/arrow/issues/35410), which is causing the certbot error.

I amended my existing PR #88 to fix this as well. I tested it out on an Ubuntu 20.04 droplet in DigitalOcean and was able to reproduce the original issue, and the latest commit I made to [my forked lemmy-ansible repo](https://github.com/christopher-conley/lemmy-ansible) deployed a lemmy instance successfully.
